packages/fxa-content-server/tests/functional/settings_v2/external_links.js

```diff
@@ -42,4 +65,50 @@ describe('external links', () => {
       'https://www.mozilla.org/en-US/about/legal/terms/services/'
     );
   });
+
+  it('renders Newsletters link in navigation when we are subscribed to a product', async () => {
+    if (process.env.CIRCLECI === 'true' && !process.env.SUBHUB_STRIPE_APIKEY) {
+      this.skip('missing Stripe API key in CircleCI run');
+    }
+
+    const email = createEmail();
+    await clearBrowserState({
+      '123done': true,
+      force: true,
+    });
+    await createUser(email, PASSWORD, { preVerified: true });
+    await openPage(ENTER_EMAIL_URL, selectors.ENTER_EMAIL.HEADER);
+    await fillOutEmailFirstSignIn(email, PASSWORD);
+    await testElementExists(selectors.SETTINGS.HEADER);
+
+    // subscribe
+    await openRP();
+    await click(selectors['123DONE'].BUTTON_SIGNIN);
+    await testElementExists(selectors.SIGNIN_PASSWORD.HEADER);
+    await click(selectors['SIGNIN_PASSWORD'].SUBMIT_USE_SIGNED_IN);
+    await testElementExists(selectors['123DONE'].AUTHENTICATED);
+    await visibleByQSA(selectors['123DONE'].BUTTON_SUBSCRIBE);
+    await click(selectors['123DONE'].LINK_LOGOUT);
+    await visibleByQSA(selectors['123DONE'].BUTTON_SIGNIN);
+    await subscribeToTestProduct();
+
+    // Signin
+    await openRP();
+    await click(selectors['123DONE'].BUTTON_SIGNIN);
+    await testElementExists(selectors.SIGNIN_PASSWORD.HEADER);
+    await click(selectors['SIGNIN_PASSWORD'].SUBMIT_USE_SIGNED_IN);
+    await testElementExists(selectors['123DONE'].AUTHENTICATED);
+    await visibleByQSA(selectors['123DONE'].SUBSCRIBED);
```

Comment:
Same thing here, this looks like a stanza that's duplicated in, sadly, a lot of spots. How about we clean this up into a functional test helper?